clash-prelude/src/Clash/Sized/Vector.hs

```diff
@@ -1676,7 +1687,7 @@ scatter = permute const
 -- <9,4,1,6,2,4>
 gather :: (Enum i, KnownNat n)
        => Vec n a  -- ^ Source vector, /xs/
-       -> Vec m i  -- ^ Index mapping, /is/
+       -> Vec m i  -- ^ SatIndex mapping, /is/
```

Comment:
These last 4 comments can just stay `Index` as they don't refer to the type `Index`, but the concept of an index